pkg/fleet-manager/pipeline/render/customTask.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright Kurator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package render
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	pipelineapi "kurator.dev/kurator/pkg/apis/pipeline/v1alpha1"
+)
+
+const (
+	CustomTaskTemplateName = "pipeline custom task template"
+)
+
+type CustomTaskConfig struct {
+	TaskName             string
+	PipelineName         string
+	PipelineNamespace    string
+	Image                string
+	Command              []string
+	Args                 []string
+	Env                  []corev1.EnvVar
+	ResourceRequirements *corev1.ResourceRequirements
+	Script               string
+	OwnerReference       *metav1.OwnerReference
+}
+
+// CustomTaskName is the name of custom task object, in case different pipeline have the same name task.
+func (cfg CustomTaskConfig) CustomTaskName() string {
+	return cfg.TaskName + "-" + cfg.PipelineName
```

Comment:
If we look at deployment->pod name generation, we may put PipelineName in front. 

And also mind the length